realm/public.gno

```diff
@@ -92,3 +94,38 @@ func RepostThread(userPostsAddr std.Address, threadid PostID, comment string) Po
 	repost := thread.AddRepostTo(caller, comment, dstUserPosts)
 	return repost.id
 }
+
+// Get posts in a thread for a user. A thread is the sequence of posts without replies.
+// While each post has an an arbitrary id, it also has an index within the thread starting from 0.
+// Limit the response to posts from startIndex up to (not including) endIndex within the thread.
+// If threadID is 0 then return the user's top-level posts. (Like render args "user".)
+// If threadID is X and replyID is 0, then return the posts (without replies) in that thread. (Like render args "user/2".)
+// If threadID is X and replyID is Y, then return the posts in the thread starting with replyID. (Like render args "user/2/5".)
+// The response includes reposts by this user (only if threadID is 0), but not messages of other
+// users that are being followed. The response is a JSON string.
+func GetThreadPosts(userPostsAddr std.Address, threadID int, replyID int, startIndex int, endIndex int) string {
+	userPosts := getUserPosts(userPostsAddr)
+	if userPosts == nil {
+		panic("posts for userPostsAddr do not exist")
+	}
+
+	if threadID <= 0 {
```

Comment:
Can we have a negative threadID?

Reply:
Posts are not stored with a negative threadID. It's possible that the function will be called where threadID is negative. We can either panic, or make the code simpler and just treat it the same as zero.

Reply:
In my opinion, if you expect that the threadID would be zero for that condition, test only for `== 0`.
And if the threadID is negative, the function will panic just after (`panic(ufmt.Sprintf("thread does not exist with id %d", threadID))`.
One common way is to validate the arguments at the top of the function and panic if threadID is negative, so it would be clearer I think (but yes, more code).

Reply:
You are right. It will panic for thread does not exist. I didn't think of that, but this is a good enough message. I changed the code to check threadID == 0. (And change replyID below.)

Reply:
Yes, and now you will be correct with the function's description in the comments.